File: quiz-app/src/types/quiz.ts
export interface Quiz {
  id: number;
  quizId: string;
  title: string;
  description: string;
}

export interface Question {
  id: number;
  questionText: string;
  options: string[];
  correctAnswer: string;
}

export interface Participant {
  id: string;
  userName: string;
  joinedAt: string;
}

File: spring-backend/src/main/java/com/quiz/model/Quiz.java
package com.quiz.model;

import jakarta.persistence.*;
import lombok.Data;
import java.util.List;

@Data
@Entity
public class Quiz {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String quizId;
    private String title;
    private String description;
    
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "quiz_id")
    private List<Question> questions;
}
